SwrveCommonSDK/src/main/java/com/swrve/sdk/SwrveImp.java

```diff
@@ -897,6 +917,7 @@ protected SwrveConversationCampaign loadConversationCampaignFromJSON(JSONObject
         return new SwrveConversationCampaign((SwrveBase<?, ?>) this, campaignData, assetsQueue);
     }
 
+    @RequiresPermission (Manifest.permission.WRITE_EXTERNAL_STORAGE)
```

Comment:
My understanding is that it requires the WRITE_EXTERNAL_STORAGE permission be declared in the android manifest. If its not then Android Studio will throw an error. Its not for runtime permission checking. I'll test this out....